src/gridcoin/researcher.cpp

```diff
@@ -1123,9 +1123,10 @@ void Researcher::RunRenewBeaconJob()
     // Do not perform an automated renewal for participants with existing
     // beacons before a superblock is due. This avoids overwriting beacon
     // timestamps in the beacon registry in a way that causes the renewed
-    // beacon to appear ahead of the scraper beacon consensus window.
+    // beacon to appear ahead of the scraper beacon consensus window. The
+    // window begins 4 hours before the next superblock by convention.
     //
-    if (!Quorum::SuperblockNeeded(pindexBest->nTime)) {
+    if (!Quorum::SuperblockNeeded(pindexBest->nTime + (60 * 60 * 4))) {
         TRY_LOCK(pwalletMain->cs_wallet, locked_wallet);
```

Comment:
This is approved, but I do have one comment. The actual scraper activation window is configurable. It defaults to 4 hours, but can be adjusted by a protocol entry, so you might want to change it to refer to the parameter instead.

unsigned int nActiveBeforeSB = 14400;

Reply:
I will put the change up.

Reply:
Ok. Missed that detail. Four hours should be fine for any configuration, but it would be better to use the explicit value I guess.

Reply:
Yep. Just in case we decide to change it in the future.